dask/array/rechunk.py

```diff
@@ -220,9 +226,20 @@ def rechunk(x, chunks="auto", threshold=None, block_size_limit=None):
 
     >>> y = x.rechunk({0: -1, 1: 'auto'}, block_size_limit=1e8)
     """
+    if n_chunks:
+        if isinstance(n_chunks, (int, float)):
+            n_chunks = (n_chunks,) * x.ndim
+
+        chunksizes = [
+            _even_chunksize(s, n_chunk) if n_chunk not in ("auto", -1) else s
```

Comment:
You might want to try `isinstance(n_chunk, str)` to support inputs like `(10, "100 MiB")`, which are allowed in other parts of this API.